### DISCUSSION TITLE
список миссий

### DISCUSSION THREAD

**Discussion by Pigzinzspace** [godot-engine-ru/shooter-3d#7] [Ideas]
**содержание документа** - последовательное описание всех миссий в игре с привязкой их к местности, входы/выходы названия локаций
шаблон 
**миссия "номер, название"**
"описание"
  
 **Краткий сюжет** - на округу спускается заражение, и все вокруг становятся зомби, один из патрульных охраны на военном заводе почему то остается человеком. 
Ему удается по рации связаться с другим выжившим, который знает  как выбраться из перекрытого завода с заблокированными лифтами и толпами зомби. Далее парень слушая советы своего нового союзника пытается выбираться из заблокированного завода.
  
**миссия 1, вызвать подмогу и выбраться из здания**
Игра начинается с катсцены:
 Человек в форме охранника обходит пустые помещения завода (патрулирует), коридор, входит в помещение цеха видит движение тени или силуэта, сообщает по рации "движение в строении 5", в ответ слышит рычание,  достает оружие подходит и видит первого зомби. 
Игрок получает управление. Игрок должен  отбиться от  первых зомби дубинкой и пистолетом, и достать из сейфа свой телефон, чтоб позвонить и доложить о проишествии. Позвонить можно только с проходной , там же герой может забрать из хранилища свой телефон, герою надо подняться на второй этаж к проходной,  дойти до лифта,  убедиться что лифт не работает, пробиться к лестнице в другом конце этажа, лестница закрыта на замок, вернуться в цех найти там клещи и выбор милишного оружия первого уровня: кувалду, разводной гаечный  ключ, цепь (можно оставить только 1 )
Дальше вернуться через цех, в котором уже опять наспавнились зомби, вскрыть замок и начать подъем по лестнице где его ждет очередная пачка зомби. 
Поднявшись по лестнице, герой двигается по коридору второго этажа в сторону холла проходной, убивая по пути зомби, на пути пост охраны завода.  Гг должен убить зомби охрану, на проходной и забрать ключи от камеры с личными вещами, зайти, убить пару зомби в хранилище. Найти свою ячейку,  достать телефон, позвонить сообщить о нападении зомби, но по телефону отвечают что сеть перегружена, герой отправляется на выход из здания и видит, что выход аварийно заблокирован, закрыта железными бронированными листами.

**миссия 2, найти другой путь**
-начинается катсцена второй миссии: 
Герой встает над селектором и настраивается на симпатичный мужской голос, который сообщает, что может перенастроиться на частоту его рации. Голос сообщает, что  он тоже из охраны, но из внутренней, зовут его "Вася",  что он в обзорной башне на верхнем этаже офисного здания, к нему не попасть, но он может помочь выбраться и привести помощь. 
Осмотрев комнату охраны герой получает доступ к запасу патронов, оружию второго уровня (находит дробовик) , аптечку. В мед шкафу также лежит запас крови, который используется как химические ловушки.
Он знает что двери заблокированы, но есть второй выход, попасть в который не так легко выход есть, открытый проход есть только через автоподаточную ленту на втором этаже в цехе автоматической сборки, но придется проехать через печь сушки, а там температура 150 градусов, поэтому лучше найти термокостюм. 
Костюм можно найти в лаборатории обработки металла на 3 этаже.  Он спускается по лестнице на третий этаж, но там железная дверь, закрытая изнутри,  клещи тут не помогут. Голос сообщает, что можно пройти через крытый коридор соседнего офисного здания (строение 4 ), соединяющий третий этаж цеха и второй этаж офиса, пройти можно если удастся открыть технический люк из тоннеля ведущего из нижнего  этажа под строением 4, а потом подняться там на второй этаж и пройти по крытому переходу. 
Мы идем в тех тоннель находим там кучу зомбяков, но без одежды и с какими-то наростами, которые прыгают как лягушки, лазают по стенам и плюются едкой дрянью. Потолки в тоннеле низкие и дальше его перегораживает решетка из прутьев так что лягушек можно загнать в угол и перебить.   Голос как не в чем не бывало сообщает что через тех тоннель можно попасть не только генератору электростанции, но  и в секретную лабораторию откуда вся эта нечисть и позбегала. И так мы нашли люк пролезли а там ничего нового,  мы пробиваемся через толпу зомбифицированного офисного планктона и наконец проходим крытый переход и видим в окна перехода, что внутреннюю часть патрулируют внушительное количество зомби, нам туда еще придется сходить, но пока мы этого  не знаем.  
Мы пришли в литейную лабораторию и нашли термокостюм, а за одно в секрете с огнеметом внутри оружие третьего уровня.

**миссия 3, путь в гораж свободен** 
кат сцена 3 миссии - герой в термокостюме сжигает несколько зомби из огнемета, сцену показывают только если герой нашел огнемет в секрете.  
Дальше герой проходит до двери, которая была закрыта изнутри  открывает засовы, которые ее запирали и возвращается на лестницу, теперь проход  в строение №3 через  сушилку открыт. Герою надо подняться на  4 этаж где северная стена строения №5(цех) соприкасается со строением №3(логистический  склад). В цехе автоматической сборки механические руки перемещают, скручивают всякие детали,  для игрока  это небольшой аркадный  пазл, и вот он стоит у ленты в печь. Голос говорит герою лечь на ленту между деталями, как следует продышаться и задержать дыхание и ждать пока выедешь с другой стороны, но если герой так сделает он погибнет, как только он начнет получать урон он должен начать бежать к выходу перепрыгивая узлы на конвейере и не касаясь боковых стен печи чтоб не поджариться, а на встречу ему выпрыгнут пару пылающих зомби и будут ему мешать пройти. Печь изнутри красная и блурит пространство вокруг. Герой пробегает печь и оказывается в строении №3. C этой стороны можно отключить сушилку чтоб не поджарится по пути назад.  Голос предлагает ему спуститься вниз, в подземный гараж где стоит готовая к отправке  военная техника, и поискать там что-то достаточно тяжелое - тягач или бронированную машину, сесть на него проехать по подземной дороге на склад готовой продукции и выбить отгрузочные ворота на склада. Герой с  четвертого этажа пробивается по центральной лестнице на первый этаж на первый уровень гаража. в гараже есть зомби, но нет никакой тяжелой техники кроме нескольких танков. Танки не заправлены, но ствол одного из них направлен в сторону закрытых ворот во дворе кластера, через которые можно добраться до заправки. Голос предлагает поискать снаряды на испытательном стенде в строении №7, что выстрелом выбить дверь во двор, и отправится за горючим. 
Строение №7 находится с востока  от цеха строения №5 там есть крытый проход на втором этаже мы возвращаемся в цех, проходим крытый проход,  проходим половину 2-го этажа, поднимаемся по лестнице и обыскиваем 3-ий там стоят всякие стрелковые  и другие стенды в частности  и для артиллерии, находим нужный калибр грузим на себя, кроме снаряда для танка мы находим на выбор дробовик или пулемет пистолет 2 оружия второго уровня (носить можно только 1 вариант) с полным боекомплектом, в секрете можно найти "прототип" оружие 0, которое позволяет "а что позволяет я пока не придумал, пока герой не понял как оно работает".  Мы возвращаемся назад идем по лестнице вниз, через крытый проход в цех дальше поднимаемся на 3 этаж на север через печь, которая к счастью не работает, далее на первый этаж логистческого склада, добираемся до  танка. 

**Миссия 4, устранить все препятствия на пути к топливу** 
катсцена миссии 4  - танк стреляет и разносит ворота гаража во внутренний двор, толпа зомби оборачивается посмотреть  кто там шумит, и также оборачивается "Объект-127", похожий на распухшего гориллоподобного  зомби из цеха но только во внешнем  экзоскелете. 
Миссия 4 проходит как в серьезном сэме, по всему двору спавнятся зомби и несутся в нашу сторону, волнами в конце на середину  перед гаражом вылетает босс. Можно с ним не сражаться но он стреляет ракетам и если они залетят в хранилище горючего, то оно рванет. Бос на дистанции обстреливает самонаводящимися ракетами которые имеют большой радиус и больно бьют, босс получает очень мало урона от атак дальнего действия, убить можно или из огнемета если гг его нашел или в мили сражаться нанося удары в коротких паузах после неудачных атак боса, мили атаки боса такие же как у  обычного моба, таран, серия ударов в одну точку, и бросок с прогиба только каждая его успешная атака убивает.  
После убийства боса. Герой перелазит через забор хранилища топлива, вкопанный в землю купола резервуаров, окруженные металлической сеткой и раздатчик топлива как на заправках, вокруг мирно пошатываясь скучает толпа зомби, которая тут же бежит к герою, Голос говорит что не стоит стрелять рядом с резервуарами,  может произойти воспламенение даже от паров. огороженная территория не такая большая, а зомби много, разобравшись с ними и открыв изнутри вход, в  ограждение, мы берем нужное топливо и возвращаемся, через двор, заправляем танк.    

**миссия 5, разблокировка**
катсецена миссии 5,- т72 заводится и спускается вниз по бетонном пандусу на подземный этаж, танк проезжает через железные двери в тоннель как через картон, двигается по подземному переезду от логистического склада до склада  отгрузки, но дорогу ему преграждают металлические столбы выходящие из под земли высотой 1.5 метра и столько же в диаметре танк останавливается. 
  Голос сообщает гг,  это значит что успели включить полную блокировку, надо перезапустить генератор, доступ к генератору через тонель из 5-го корпуса заблокирован, но есть еще один тех проход  который ведет в тоннель между секертной лабораторией и генератором, прямо из того места где гг находится  сейчас. 
  Герой спускается на 1.5 метра вниз через люк в коридор между генератором и подземной лабораторией, проход к генератору так же заблокирован решеткой как и со стороны корпуса 5. Голос говорит что можно пройти в секретную лабораторию и перезагрузить там сервера, это разблокрует вход в генераторную.  Герой проходит тоннель убивает там несколько зомби лягушек, заходит в здание секретной лаборатории, там КПП внутренней охраны и зомби охранники. Убивает много зомби охоранников находит несколько экземпляров 1,2,3,4 оружия и 2-3 крио гранаты, полный боезапас оказывается на 6-ом этаже строения 1. Это главный этаж, холл, вдоль стен колбы с жидкостью со странными фигурами полу людей мутантов внутри, с подключенными трубками и шлангами, стоящими стойками с аппаратурой, то ли научной то ли медицинской. Существа в колбах то ли спят, то ли заспиртованы. На одной стене TV-панель как  в центре управления полетами с рядами цифр и столбиковыми диаграммами. Площадь комнаты заставлена  столами с мониторами стульями и большим количеством зомби в белых и зеленых, как у врачей халатах, которые покачиваются в полудреме,но начинают просыпаться и бежать в сторону героя. После зачистки, Голос говорит найти зомби с бейджиком зеленого цвета, это гостевая магнитная карта с правом прохода во внутреннюю часть лаборатории, если не зеленой бери белую, голос говорит иди к центральном коридору с другой стороны от входа  там лифт на 7 этаж. Герой убивает зомби перед лифтом, в лифте. После лифта его ждет здоровяк в экзоскелете, как босс 127, только без ракет. Коридор, в обе стороны коридора вход лабораторию, с одной стороны зала офисные кабинеты внутри которых зомби с зелеными синими и красными баджиками. по этажу бегают   гугл собаки (или летают коптеры), которые появляются постоянно неоткуда их можно сбить одним ударом любого оружия,  собаки бьют электрическими разрядами, получив 4 разряда подряд герой на несколько секунд теряет способность двигаться и падает.  Если подобрать зеленый бейджик собаки перестают атаковать. В конце коридора вход в дата центр он закрыт дверь закрыта ключ на столе в одном из кабинетов. Герой убивает зомби в дата центре, проходит к серверу и отрубает его, Голос говорит "Отлично, теперь я могу рассказать тебе кто я на самом деле". Голос "рассказывает кто он на сама деле".   
  **рассказ Братишки Зомби**
На самом деле он не охранник, а зараженный, но заражен уже давно из за того, что он не потерял рассудок его держали тут и исследовали по особой программе, так как он и сам был ученым ему давали работать, естественно под надзором. Он не причина эпидемии и не знает кто причина те кто его держали в плену и исследовали превратились в зомби вместе со всеми, так что наверно не они.  Его заманили в ловушку которая была спланирована заранее пару раз при нем разные люди обмолвились что материалы о регрессивной терапии хранятся на 3 этаже лабы, естественно он рванул туда как только случилась катастрофа, ведь это то что его интересует больше всего. Это оказалась ловушка как только он начал копаться в данных  искать информацию на тему  регрессивной терапии его комната была заблокирована. На счастье  он подобрал у погибшего охранника рацию, и когда выяснилось что есть не зараженный попробовал использовать его чтоб выбраться из ловушки но и гг же он тоже помог. И вообще они с гг братья, родственники по иммунитету, правда у него от человека только мозги так что он младший брат, поэтому  гг может звать его Братишка Зомби. 
 
  
**Дополнение к миссии 5** с одной стороны коридора находятся офисы, с другой стороны коридора находятся 4 крупных помещения с большими дверями и одно с обычной дверью, чтоб выполнить миссию их не надо вскрывать, у одного из помещений  рядом с дверью выломан кусок стены на его двери написано "объект-119" как маркировка на экзоскелете здоровяка которого мы убили на выходе из лифта, двери  помещения "объект - 127" открыты  и там нет следов разрушений. Остается еще 2 больших помещения и одно маленькое, над маленьким надпись "Объект-0514", оно закрыто на обычную дверь которую можно выломать ударом ноги. В помещение  "Объект-0514" находятся стойки с книгами лабораторные столы с научным оборудованием , телевизор с видео приставкой, другие атрибуты жилого помещения. 
Большие помещения разделены на 2 части прозрачным а одну сторону стеклом, как в чужом или допросные комнаты в американских фильмах, одна часть это лаборатория с оборудованием, вторая представляет из себя вальер с крючьями и цепями и видимо для фиксации объектов.  В Помещения "объект-307" и "объект-097" вход закрыт,  там находятся босы. Попасть туда герой может только если нашел секреты с взрывчаткой по ходу игры, тогда можно взорвать двери. Больше никак открыть их нельзя. После зачистки комнат от боса и зомби, герой может найти в комнате  оружие 5 "кинетическое ружье"  и комплект "компаньон робопес" гаджет который можно активировать на себя и превратить в питомца, иногда замедляющего ближайших врагов.

 **миссия 6, зачистка**
 Кат сцена миссии 6 - Герой вылазит из люка, на броне танка сидит "объект-0514" и приветственно машет ему рукой, столбики блокировки по прежнему подняты вверх, герой покачивает головой в том плане что "ну ты и фрукт!".
 Братишка Зомби говорит, чтоб разблокировать, систему защиты у него уйдет время, а пока можно заняться полезным делом, сжечь подземную лабораторию и всех кто там еще остался. Пока ты занимаешься подготовкой он вытащит из лабы все что может пригодиться и отрубит внутри лабы систему пожаротушения. 
 Герою нужно собрать удлинитель топливо-вода из пожарных шлангов, для этого он бегает по территории завода  и собирает их в пожарных шкафах, убивая толпы зомби, когда он приносит шланг к заправке  Братишка Зомби называет следующий пожарный шкаф, в конце концов герой пробежав изрядно комнат и покрошив изрядно зомбей собирает топливо-вод. 
 Кат сцена 2 миссии 6 -  бензин перестает течь в вентиляцию, Герой подносит рацию к уху, Братишка Зомби говорит - "Ты закончил с топливом, окей у меня тоже все готово, я выбрался можешь начинать. На выход из гаража я притащил все найденное в лабе оружие и патроны. Там внутри повсюду запах бензина, кек." Герой поджигает и бросает в вентиляцию коктейль Молотова. 
Из вентиляции валит черный столб дыма и начинают вылетать горящие и не горящие зомби, некоторые вырываются прямо из под земли  и все лезут на героя, появляются и все виды босов 127,119,307 и 097 , босы спавнятся парами, все мобы спавнятся по времени кроме ситуации с лимитом если лимит достигнут то мобы ждут в очереди. 

Финальная катсцена эпизода  - герой загружает припасы в фургон типо буханки, Голос Братишки Зомби на заднем плане говорит  "меня зомби не трогают так что останусь тут и продолжу исследования, может удастся найти вакцину или какой то вирус который поражает зараженных, хотя я этим занимался в лабе последние 5 лет, так что вряд ли будет молниеносный прорыв, еще позабочусь чтоб тут холодильники работали и был всегда запас еды и чистой воды для тебя. Желаю удачи в поисках не зараженных, но если заражение было не случайным, а намеренным то площадь поражения может быть очень большой, таких лаб как ту что мы сожгли насколько я знаю несколько только в нашем районе есть наверно и в разных городах. Береги себя братишка!"  Герой прыгает в буханку двери гаража открываются  впереди простирается огромная территория завода.  Играет рэпчина, буханка уезжает вдаль, поднимая пыль, идут титры.

Конец!

Не, еще не конец.
**секретная подземная лаба**  это особый аттракцион для игрока рядом с лифтом с 6-го на 7-ой есть еще 1 лифт на котором 2 кнопки вниз и выход, выход возвращает на 7-ой а вниз опускает на этажи с пятого по первый последовательно, только дело в том что их там не 5, а бесконечное количество. 
Игрок выходит из лифта в короткий коридор, в котором стоят автомат с газировкой из хл, в котором игрок может пополнить ХП и  оружейный шкаф из комнаты охраны в котором лежит абсолютно все виды стандартного оружия и полный боезапас. 
в конце коридора находится вход на один из игровых этажей, любых зданий зоны (спавнится рандомная локация завода) 
на эту локацию загружается случайный сбалансированный набор монстров которые спавнятся по углам комнаты  в несколько волн  и бегут убивать героя.
Если герой убил всех монстров и прошел комнату, он может пройти в лифт и проехать в следующую комнату или выйти.
Если игрок пройдя несколько уровней и выходит от туда во время сюжета, то его ХП, патроны и оружие становятся такими как были когда он вошел.  
После окончания первого эпизода в главном меню появляется кнопка "лаба", которая загружает игрока в этот лифт.  
Ходят слухи что на 128304 этаже лабы  есть проход на секретный уровень.

**Comment by Pigzinzspace** [comment]
![image](https://user-images.githubusercontent.com/79694583/152012084-f0953a99-99ee-4b92-9e3c-eadcfaee6f74.png)

0 - генераторная под землей
1 - секретная лаба под землей
2 - склад готовой продукции
3 - логистический склад
4 - офисы и мелкая сборка 
5 - цех тяжелого оборудования
6 - оружейный склад
7 - мастерские, отдел качество, отдел новой продукции, испытельные стенды
8 - хранилище горючего 
9 - столовая, госпиталь, актовый зал, социальные институты ...

**Comment by soundsbeard** [comment]
> миссия 1
> кат сцена заканчивается открывается меню перса с возможностью назначить сложность игры или выбрать опции (если они будут)
дальше игра сохраняется и игрок получает управление.

зачем такие сложности? может стоит сделать классически, ты перед игрой выбираешь сложность, а потом уже игра и всё остальное?
так получается, что надо будет сначала загрузить родительскую сцену, потом туда подгрузить ui с выбором сложности, и накинуть эту сложность на уже работающую сцену.

**Reply by slapin** [reply]
Ну на самом деле и так и так можно но явный выбор сложности внутри миссии это как-то так себе, лучше тогда методом выбора плана действий (абстрагируясь) сложность 1 - спасаем принцессу - залезая в окно с вертолета, 2 - спасаем принцессу, залезая по стене замка и отстреливаясь, 3 - спасаем принцессу, продираясь через весь замок и типа в зависимости от выбранной сложности мы получаем больше ништяков для прокачки гг и базы и покупки оружия...

Выбор миссии думаю лучше делать из базы и возможно выбор сложности должен обыгрываться как "планирование" миссии. Это будут более-менее общие системы, так будет их проще переиспользовать для всех миссий.

**Comment by soundsbeard** [comment]
> в гараже есть зомби нет бульдозера, но зато там есть танк.

я считаю это плохой идеей.
пока читал, танк мне показался испанской инквизицией (в плохом смысле), я его никак не ожидал.
мне кажется, это совршенно не вписывается в картину.

в ответ предлагаю:
если есть необходимость выбить дверь, то предлагаю "слизать" сцену из хл2, эпизод на лодке, надо было выбить ворота висящей перекладиной: https://youtu.be/jElU1mD8JnI?t=2688

то есть вместо гаража, это будет что-то типа ангара починки техники, где за троссы будет привязана какая-то очень крупная деталь прямо около ворот.
надо будет срезать/оборвать нужные троссы, чтобы эта деталь "маятником" выбила дверь с шумом. и этот шум логически впишется в твоё дальнейшее повествование.

**Reply by Pigzinzspace** [reply]
вообще то  это военный завод тяжелой техники и танку на складе готовой продукции самое место, он там просто прописан, ты заходишь в гараж где стоитя военные грузовики, грады, джипы БМП и пара танков, и отсюда отгружают.
видимо я недостаточно четко все описал.  Но это гараж куда он полез искать военный тягач или бульдозер именно гараж готовых военных  машин.

**Reply by slapin** [reply]
Садисты, вы пытаетесь заставить игрока думать...

**Comment by alex7850** [comment]
Я бы выделил главную цель миссии. Например "добраться и активировать что-то"
Сложно продираться через текст, поэтому и комментировать пока не берусь

**Reply by slapin** [reply]
многостадийные миссии рулят, особенно с ветвлением если зафейлил один способ пройти на тебе альтернативный...

**Reply by Pigzinzspace** [reply]
так и получается что в начале миссии ты не особо в курсе с чем столкнешся  знаешь только общее направление.

**Reply by alex7850** [reply]
я про текст на гитхабе, а не в игре. Сложно читать без минимального форматирования

**Reply by Pigzinzspace** [reply]
я написал название каждой миссии, у каждой есть название.

**Comment by me2beats** [comment]
> достать телефон, позвонить сообщить о нападении зомби, получить ответ - чтоб перестал баловаться

Как-то звучит не очень убедительно, либо как очень большое разгильдяйство со стороны тех кому звонит.
Чем мотивирован такой ответ?
Либо просто что гг (охранник)такой несерьёзный шалопай и уже были подобные случаи шуток с его стороны, что всё равно странно
То есть гг мог бы по телефону просто сказать: у нас ЧП, срочно выезжаете.
И должны выехать или проверить позвонив другим хотя бы.
Ну мне так представляется.

Что за телефон вообще, он с камерой/интернетом?

**Reply by Pigzinzspace** [reply]
ну да это не реалистично но это задумывалось как шутка типо, - товарищ полковник докладываю город атакуют зомби, - мальчик я тебе сейчас приеду и  уши надеру, а твою маму ...

если это прям совсем кринж, то можно например сделать  что сеть не доступна, но тогда будет раньше времени понятно что в городе все тоже накрылось

**Reply by slapin** [reply]
Ну можно обыграть классику типа слушай чувак звони вот по такому номеру... звонишь на него а там тебе "психоневрологический диспанцер оператор Колян слушает" это будет референс на вполне реальную историю.

**Reply by Pigzinzspace** [reply]
это субкультурный мем, я например его не знаю

**Comment by me2beats** [comment]
Компаньон мужчина, тоже охранник?
Что насчёт компаньона-женщины, научной работницы.

**Reply by soundsbeard** [reply]
а женскую озвучку где взять?

**Reply by Pigzinzspace** [reply]
ну что так прям плохо сделать 2 мужских перса ?

**Reply by slapin** [reply]
Вы озвучку хотите? я думаю что она найдется если набор фраз найти и поискать озвучивальщиков на фрилансе, а по началу можно использовать синтезируемые голоса.

**Reply by Pigzinzspace** [reply]
да у нас есть брутальный мужской голос Кирю попросим

**Comment by me2beats** [comment]
Также, есть идея что можно несколько миссий играть за женщину.
Они могут друг другу помогать выбраться
В каких-то миссиях они возможно могут быть и вместе.

**Reply by slapin** [reply]
герои должны взаимодействовать, но тут уже скользкая дорожка смены гг которая может разработку очень сильно усложнить...

**Reply by Pigzinzspace** [reply]
мне тоже не нравится идея перспективой взаимодействия, смены ролей  и еще  кома лора и социалочки
